packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -194,7 +202,14 @@ export class AwsCustomResource extends cdk.Construct implements iam.IGrantable {
       for (const statement of props.policyStatements) {
         provider.addToRolePolicy(statement);
       }
-    } else { // Derive statements from AWS SDK calls
+    } else {
+
+      if (!props.allowStarPermissions) {
+        throw new Error('`allowStarPermissions` must be set to true when '
```

Comment:
This change basically means that you either have to specify policyStatements or generateStarPermissions, which means that perhaps the “union” pattern can be applied:

```ts
policy: ProviderPolicy.custom(...statements)
policy: ProviderPolicy.deduceFromSdkCalls({ allowStarPermissions: true })
```

Or something like that. This might also be a bit more future compatible as we can add additional strategies.

Reply:
I like it.  Im thinking we don't even need the `allowStarPermissions` at all. The user opts in to star permissions by choosing to use this method. Maybe we can name it `generateStarPermissions` to make it clearer, but i think the doc might even be good enough here.

Reply:
Hmm, not so simple. Let me come up with something.

Reply:
Outdated - resolving